src/TargetPlugin.groovy

```diff
@@ -28,22 +32,5 @@ class TargetPlugin implements TerraformPlanCommandPlugin, TerraformApplyCommandP
     }
 
     @Override
-    public void apply(TerraformEnvironmentStage stage) {
-        stage.decorate(ALL, addBuildParams())
-    }
-
-    public static Closure addBuildParams() {
-        return { closure ->
-            def params = [
-                string(name: 'RESOURCE_TARGETS', defaultValue: '', description: 'comma-separated list of resource addresses to pass to plan and apply "-target=" parameters'),
-            ]
-            def props = [
-                parameters(params)
-            ]
-            properties(props)
-
-            closure()
-        }
-    }
```

Comment:
I think I actually prefer this implementation. It doesn't require the Plugin to know about the internal structure of the TerraformEnvironmentStage.

Reply:
With this implementation, I cannot append to the parameters. So any plugins that add parameters will overwrite eachother.